<a href="https://colab.research.google.com/github/sheema84/Omnifood/blob/master/MOBILE%20PRICE%20PREDICTION%20USING%20ML.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**DATA SCIENCE JUNE MINOR PROJECT**<br>
**DONE BY: SHEEMA ANUSH**

PROBLEM STATEMENT:
 Create a classification model to predict whether price range of
mobile based on certain specifications


CONTEXT:
 An entrepreneur has started his own mobile company. He wants to give
tough fight to big companies like Apple, Samsung etc.
He does not know how to estimate price of mobiles his company creates. In this
competitive mobile phone market, one cannot simply assume things. To solve this
problem, he collects sales data of mobile phones of various companies.
He wants to find out some relation between features of a mobile phone (e.g., RAM,
Internal Memory etc) and its selling price. But he is not so good at Machine Learning.</BR></BR>
So, he needs your help to solve this problem.
In this problem you do not have to predict actual price but a price range indicating
how high the price is


STEPS TO CONSIDER:</BR>
1)Remove handle null values (if any).</BR>
2)Split data into training and test data.</BR>
3)Apply the following models on the training dataset and generate the predicted value for
the test dataset</BR>
a) Logistic Regression</BR>
b) KNN Classification</BR>
c) SVM Classifier with linear and rbf kernel</BR>
4)Predict the price range for test data</BR>
5)Compute Confusion matrix and classification report for each of these models.</BR>
6)Report the model with the best accuracy</BR>

**LOGISTIC REGRESSION**

In [ ]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns

In [ ]:
df=pd.read_csv('/content/mobile_price_range_data.csv')
df.head()

battery_power  blue  clock_speed  dual_sim  fc  four_g  int_memory  m_dep  \
0            842     0          2.2         0   1       0           7    0.6   
1           1021     1          0.5         1   0       1          53    0.7   
2            563     1          0.5         1   2       1          41    0.9   
3            615     1          2.5         0   0       0          10    0.8   
4           1821     1          1.2         0  13       1          44    0.6   

   mobile_wt  n_cores  ...  px_height  px_width   ram  sc_h  sc_w  talk_time  \
0        188        2  ...         20       756  2549     9     7         19   
1        136        3  ...        905      1988  2631    17     3          7   
2        145        5  ...       1263      1716  2603    11     2          9   
3        131        6  ...       1216      1786  2769    16     8         11   
4        141        2  ...       1208      1212  1411     8     2         15   

   three_g  touch_screen  wifi  price_range  
0        0             0     1            1  
1        1             1     0            2  
2        1             1     0            2  
3        1             0     0            2  
4        1             1     0            1  

[5 rows x 21 columns]

In [ ]:
df.isnull().sum()

battery_power    0
blue             0
clock_speed      0
dual_sim         0
fc               0
four_g           0
int_memory       0
m_dep            0
mobile_wt        0
n_cores          0
pc               0
px_height        0
px_width         0
ram              0
sc_h             0
sc_w             0
talk_time        0
three_g          0
touch_screen     0
wifi             0
price_range      0
dtype: int64

In [ ]:
df.shape

(2000, 21)

In [ ]:
df.dtypes

battery_power      int64
blue               int64
clock_speed      float64
dual_sim           int64
fc                 int64
four_g             int64
int_memory         int64
m_dep            float64
mobile_wt          int64
n_cores            int64
pc                 int64
px_height          int64
px_width           int64
ram                int64
sc_h               int64
sc_w               int64
talk_time          int64
three_g            int64
touch_screen       int64
wifi               int64
price_range        int64
dtype: object

In [ ]:
x=df.iloc[:,:-1]
y=df.iloc[:,-1]
print(x.shape)
print(y.shape)

(2000, 20)
(2000,)


In [ ]:
from sklearn.model_selection import train_test_split
x_tr,x_te,y_tr,y_te=train_test_split(x,y,test_size=0.25)
print(x_tr.shape)
print(x_te.shape)
print(y_tr.shape)
print(y_te.shape)

(1500, 20)
(500, 20)
(1500,)
(500,)


In [ ]:
from sklearn.linear_model import LogisticRegression

In [ ]:
reg=LogisticRegression()
reg.fit(x_tr,y_tr)

/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_logistic.py:818: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG,


LogisticRegression()

In [ ]:
print("Training score",reg.score(x_tr,y_tr))
print("Testing score",reg.score(x_te,y_te))

Training score 0.638
Testing score 0.628


In [ ]:
ypred=reg.predict(x_te)
print(ypred)

[2 0 0 2 2 1 2 1 3 0 0 2 0 3 1 3 2 1 3 3 1 3 3 3 3 0 3 3 2 2 2 3 0 1 2 1 0
 3 2 3 2 2 1 1 2 3 0 1 3 0 2 3 2 0 2 1 1 2 3 0 2 3 0 2 3 2 0 3 3 3 2 0 1 2
 2 0 1 3 0 1 1 2 0 0 1 0 1 2 0 1 2 2 3 3 2 2 0 0 3 0 1 2 3 1 1 2 0 0 0 2 1
 3 1 0 3 1 0 1 0 2 2 0 1 0 3 0 2 1 1 0 0 3 2 3 2 1 2 3 3 0 2 1 3 1 1 2 1 1
 1 3 2 2 2 3 0 2 3 2 2 0 0 2 1 1 3 3 1 0 3 2 2 1 0 1 0 3 3 0 0 3 2 2 3 3 0
 3 0 3 3 1 2 3 3 2 3 0 0 1 3 3 3 0 0 1 3 2 3 2 1 3 1 0 0 3 3 1 2 3 2 3 2 2
 0 2 2 0 3 0 2 0 3 3 3 3 1 0 0 3 1 1 1 1 0 1 0 1 3 0 0 0 2 2 0 2 0 1 3 3 0
 0 0 0 3 1 2 0 2 3 2 2 3 3 0 3 1 1 0 2 3 3 0 0 0 1 1 0 3 0 0 3 1 2 0 2 3 2
 2 1 3 2 2 2 0 2 3 2 0 2 3 2 1 3 3 3 1 3 3 3 2 1 1 3 0 1 2 1 3 2 3 2 1 2 2
 3 1 1 3 1 3 0 3 1 2 3 3 1 1 1 0 3 3 3 0 1 0 1 1 3 1 2 1 3 2 0 3 0 3 2 3 0
 1 1 1 2 3 2 2 3 2 2 3 0 0 3 2 1 2 3 3 3 3 0 3 3 3 1 0 0 2 1 3 2 3 2 1 0 3
 1 3 1 2 1 3 3 0 0 0 3 3 2 3 3 0 2 1 0 2 0 1 0 1 0 1 2 2 3 3 3 1 2 1 3 0 0
 2 3 2 2 0 2 0 2 2 1 3 1 1 0 0 3 2 3 2 2 2 3 2 1 3 3 0 1 2 1 2 1 0 0 0 1 1
 3 1 1 0 3 2 0 3 0 2 3 0 

In [ ]:
x_te['Actual Values']=y_te
x_te['Predicted Values']= ypred
print(x_te)

      battery_power  blue  clock_speed  dual_sim  fc  four_g  int_memory  \
609            1212     0          2.6         0   1       1          46   
567            1269     1          1.3         0  14       0          52   
1995            794     1          0.5         1   0       1           2   
1422           1109     0          2.1         0   1       0          34   
1958           1976     1          2.6         1   6       0          45   
...             ...   ...          ...       ...  ..     ...         ...   
161             961     1          1.5         1  10       0          51   
1138           1413     1          1.7         0   2       0          10   
1661           1902     0          0.5         0  10       0          13   
151             742     0          2.2         1   4       0          56   
652            1462     1          1.7         1   1       0          25   

      m_dep  mobile_wt  n_cores  ...  px_width   ram  sc_h  sc_w  talk_time  \
609     

In [ ]:
from sklearn.metrics import confusion_matrix, classification_report

In [ ]:
cm=confusion_matrix(y_te,ypred)
print(cm)

[[ 98  22   4   0]
 [ 21  58  28   3]
 [  0  29  58  41]
 [  0   1  37 100]]


In [ ]:
print(classification_report(y_te,ypred))

              precision    recall  f1-score   support

           0       0.82      0.79      0.81       124
           1       0.53      0.53      0.53       110
           2       0.46      0.45      0.45       128
           3       0.69      0.72      0.71       138

    accuracy                           0.63       500
   macro avg       0.63      0.62      0.62       500
weighted avg       0.63      0.63      0.63       500



**ACCURACY OF LOGISTIC REGRESSION MODEL: 0.63**

**KNN CLASSIFICATION**

In [ ]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns

In [ ]:
df=pd.read_csv('/content/mobile_price_range_data.csv')
df.head()

battery_power  blue  clock_speed  dual_sim  fc  four_g  int_memory  m_dep  \
0            842     0          2.2         0   1       0           7    0.6   
1           1021     1          0.5         1   0       1          53    0.7   
2            563     1          0.5         1   2       1          41    0.9   
3            615     1          2.5         0   0       0          10    0.8   
4           1821     1          1.2         0  13       1          44    0.6   

   mobile_wt  n_cores  ...  px_height  px_width   ram  sc_h  sc_w  talk_time  \
0        188        2  ...         20       756  2549     9     7         19   
1        136        3  ...        905      1988  2631    17     3          7   
2        145        5  ...       1263      1716  2603    11     2          9   
3        131        6  ...       1216      1786  2769    16     8         11   
4        141        2  ...       1208      1212  1411     8     2         15   

   three_g  touch_screen  wifi  price_range  
0        0             0     1            1  
1        1             1     0            2  
2        1             1     0            2  
3        1             0     0            2  
4        1             1     0            1  

[5 rows x 21 columns]

In [ ]:
df['price_range'].value_counts()

1    500
2    500
3    500
0    500
Name: price_range, dtype: int64

In [ ]:
df.isnull().sum()

battery_power    0
blue             0
clock_speed      0
dual_sim         0
fc               0
four_g           0
int_memory       0
m_dep            0
mobile_wt        0
n_cores          0
pc               0
px_height        0
px_width         0
ram              0
sc_h             0
sc_w             0
talk_time        0
three_g          0
touch_screen     0
wifi             0
price_range      0
dtype: int64

In [ ]:
from sklearn.model_selection import train_test_split

x_tr,x_te,y_tr,y_te=train_test_split(x,y,test_size=0.25)

print(x_tr.shape)
print(x_te.shape)
print(y_tr.shape)
print(y_te.shape)

(1500, 20)
(500, 20)
(1500,)
(500,)


In [ ]:
from sklearn.neighbors import KNeighborsClassifier

In [ ]:
m1=KNeighborsClassifier(n_neighbors=19)
m1.fit(x_tr,y_tr)

KNeighborsClassifier(n_neighbors=19)

In [ ]:
print("Training score",m1.score(x_tr,y_tr))
print("Testing score",m1.score(x_te,y_te))

Training score 0.946
Testing score 0.952


In [ ]:
ypred1=m1.predict(x_te)
print(ypred1)

[3 3 2 1 2 0 2 0 0 2 1 1 3 1 3 3 1 2 3 1 0 0 1 3 2 1 2 1 3 0 3 3 2 2 2 1 2
 0 3 2 1 1 0 3 2 1 2 3 3 1 1 2 3 2 1 3 3 2 2 0 0 0 1 2 3 3 0 3 1 3 2 2 1 2
 3 3 3 2 0 3 0 2 2 2 3 2 0 0 3 0 2 2 2 1 2 2 1 3 1 0 3 0 2 0 0 0 3 0 3 2 3
 1 3 0 0 3 0 2 0 3 1 1 3 2 0 3 2 0 3 0 1 2 0 2 0 0 0 0 2 0 3 0 2 3 1 0 3 2
 3 3 1 0 0 2 3 1 0 3 2 2 0 1 2 1 1 2 1 0 3 2 3 0 3 0 1 1 0 2 2 2 3 3 1 3 1
 2 3 3 0 0 3 2 3 1 0 2 2 0 3 1 0 0 1 3 2 0 0 0 0 1 2 0 1 0 1 3 2 2 0 3 3 0
 3 1 0 3 3 0 0 3 0 0 1 1 0 0 2 2 3 3 0 3 3 1 3 3 2 1 2 1 1 0 0 3 0 1 3 3 3
 2 2 1 0 0 3 0 0 2 1 2 2 1 0 0 3 3 3 3 0 3 3 2 2 0 3 0 1 2 3 3 1 0 0 1 0 3
 2 2 1 0 2 0 3 1 0 3 0 0 1 3 1 1 1 0 2 2 2 2 3 1 0 3 1 2 2 0 3 3 0 2 0 2 3
 1 3 1 0 1 2 1 3 0 0 0 1 1 3 1 2 0 1 0 2 3 2 3 2 3 2 1 2 0 3 2 2 3 3 2 2 3
 0 2 3 2 0 0 1 0 3 0 1 2 1 0 3 2 3 0 3 0 0 0 3 0 1 2 1 3 3 2 1 0 0 1 3 1 2
 2 2 2 3 0 2 1 2 1 0 0 0 3 1 3 2 2 1 3 3 1 0 2 3 3 1 0 1 3 2 3 1 3 2 2 3 3
 2 1 3 1 2 3 1 0 2 0 1 1 3 3 3 1 2 1 0 2 2 0 2 3 0 3 1 0 2 0 1 1 1 0 0 0 1
 1 2 0 2 2 0 0 0 1 0 2 1 

In [ ]:
cm1=confusion_matrix(y_te,ypred1)
print(cm1)

[[137   2   0   0]
 [  1 101   5   0]
 [  0   3 115   8]
 [  0   0   5 123]]


In [ ]:
print(classification_report(y_te,ypred1))

              precision    recall  f1-score   support

           0       0.99      0.99      0.99       139
           1       0.95      0.94      0.95       107
           2       0.92      0.91      0.92       126
           3       0.94      0.96      0.95       128

    accuracy                           0.95       500
   macro avg       0.95      0.95      0.95       500
weighted avg       0.95      0.95      0.95       500



**ACCURACY USING KNN CLASSIFICATION MODEL: 0.95**

**SBM CLASSIFIER WITH RBF AND LINEAR KERNEL**

In [ ]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns

In [ ]:
df=pd.read_csv('/mobile_price_range_data.csv')
df.head()

battery_power  blue  clock_speed  dual_sim  fc  four_g  int_memory  m_dep  \
0            842     0          2.2         0   1       0           7    0.6   
1           1021     1          0.5         1   0       1          53    0.7   
2            563     1          0.5         1   2       1          41    0.9   
3            615     1          2.5         0   0       0          10    0.8   
4           1821     1          1.2         0  13       1          44    0.6   

   mobile_wt  n_cores  ...  px_height  px_width   ram  sc_h  sc_w  talk_time  \
0        188        2  ...         20       756  2549     9     7         19   
1        136        3  ...        905      1988  2631    17     3          7   
2        145        5  ...       1263      1716  2603    11     2          9   
3        131        6  ...       1216      1786  2769    16     8         11   
4        141        2  ...       1208      1212  1411     8     2         15   

   three_g  touch_screen  wifi  price_range  
0        0             0     1            1  
1        1             1     0            2  
2        1             1     0            2  
3        1             0     0            2  
4        1             1     0            1  

[5 rows x 21 columns]

In [ ]:
df.isnull().sum()

battery_power    0
blue             0
clock_speed      0
dual_sim         0
fc               0
four_g           0
int_memory       0
m_dep            0
mobile_wt        0
n_cores          0
pc               0
px_height        0
px_width         0
ram              0
sc_h             0
sc_w             0
talk_time        0
three_g          0
touch_screen     0
wifi             0
price_range      0
dtype: int64

In [ ]:
df.dtypes

battery_power      int64
blue               int64
clock_speed      float64
dual_sim           int64
fc                 int64
four_g             int64
int_memory         int64
m_dep            float64
mobile_wt          int64
n_cores            int64
pc                 int64
px_height          int64
px_width           int64
ram                int64
sc_h               int64
sc_w               int64
talk_time          int64
three_g            int64
touch_screen       int64
wifi               int64
price_range        int64
dtype: object

In [ ]:
x=df.iloc[:,:-1]
y=df.iloc[:,-1]
print(x.shape)
print(y.shape)

(2000, 20)
(2000,)


In [ ]:
x.head()

battery_power  blue  clock_speed  dual_sim  fc  four_g  int_memory  m_dep  \
0            842     0          2.2         0   1       0           7    0.6   
1           1021     1          0.5         1   0       1          53    0.7   
2            563     1          0.5         1   2       1          41    0.9   
3            615     1          2.5         0   0       0          10    0.8   
4           1821     1          1.2         0  13       1          44    0.6   

   mobile_wt  n_cores  pc  px_height  px_width   ram  sc_h  sc_w  talk_time  \
0        188        2   2         20       756  2549     9     7         19   
1        136        3   6        905      1988  2631    17     3          7   
2        145        5   6       1263      1716  2603    11     2          9   
3        131        6   9       1216      1786  2769    16     8         11   
4        141        2  14       1208      1212  1411     8     2         15   

   three_g  touch_screen  wifi  
0        0             0     1  
1        1             1     0  
2        1             1     0  
3        1             0     0  
4        1             1     0

In [ ]:
y.head()

0    1
1    2
2    2
3    2
4    1
Name: price_range, dtype: int64

In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
x_tr,x_te,y_tr,y_te=train_test_split(x,y,test_size=0.25)

print(x_tr.shape)
print(x_te.shape)
print(y_tr.shape)
print(y_te.shape)

(1500, 20)
(500, 20)
(1500,)
(500,)


In [ ]:
from sklearn.svm import SVC

In [ ]:
s=SVC(kernel='rbf',C=1,gamma=0.001)
s.fit(x_tr,y_tr)

SVC(C=1, gamma=0.001)

In [ ]:
print("Training score",s.score(x_tr,y_tr))
print("Testing score",s.score(x_te,y_te))

Training score 1.0
Testing score 0.24


In [ ]:
ypred=s.predict(x_te)
print(ypred)

[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 3 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 

In [ ]:
from sklearn.metrics import confusion_matrix, classification_report

In [ ]:
cm=confusion_matrix(y_te,ypred)
print(cm)

[[  0 123   0   0]
 [  0 118   0   0]
 [  0 134   1   0]
 [  0 123   0   1]]


In [ ]:
print(classification_report(y_te,ypred))

              precision    recall  f1-score   support

           0       0.00      0.00      0.00       123
           1       0.24      1.00      0.38       118
           2       1.00      0.01      0.01       135
           3       1.00      0.01      0.02       124

    accuracy                           0.24       500
   macro avg       0.56      0.25      0.10       500
weighted avg       0.57      0.24      0.10       500



/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


**ACCURACY WITH SVM(RBF KERNEL)=0.24**

In [ ]:
s1=SVC(kernel='linear',C=1)
s1.fit(x_tr,y_tr)

SVC(C=1, kernel='linear')

In [ ]:
print("Training score",s1.score(x_tr,y_tr))
print("Testing score",s1.score(x_te,y_te))

Training score 0.9906666666666667
Testing score 0.972


In [ ]:
ypred_1=s1.predict(x_te)
print(ypred_1)

[3 0 1 0 0 2 3 1 3 0 0 0 3 3 2 2 0 3 0 2 2 2 1 1 1 0 2 2 3 0 0 1 1 1 3 0 2
 2 2 0 1 3 3 3 2 0 0 1 2 3 0 2 3 0 1 2 0 2 2 0 3 1 3 3 1 3 1 2 2 2 2 1 1 2
 3 3 2 1 0 0 3 2 2 2 1 3 0 1 2 1 3 2 2 3 1 3 2 3 1 2 2 3 3 0 3 1 0 0 3 1 3
 0 1 2 0 3 1 2 1 0 2 2 1 1 3 2 3 2 3 1 2 0 0 3 2 3 1 2 0 2 0 0 3 2 1 1 1 0
 3 2 0 3 3 0 1 3 3 1 1 2 1 3 0 0 0 2 2 1 3 0 2 1 3 3 0 1 1 2 3 2 2 0 3 1 3
 1 0 3 2 1 2 3 3 1 3 0 1 1 3 0 1 0 0 3 0 2 3 1 1 3 1 3 2 3 1 2 3 2 0 1 2 3
 0 1 1 2 0 1 0 1 3 2 2 2 0 2 3 0 0 2 0 0 3 3 0 0 0 1 1 0 1 2 2 0 1 3 2 0 3
 3 1 1 2 3 1 0 3 3 2 0 0 3 2 2 0 3 3 1 0 1 2 3 1 1 3 1 2 1 2 0 2 2 2 3 0 1
 0 2 0 3 0 2 3 3 1 2 0 0 1 0 3 2 3 0 2 3 2 0 2 1 0 3 3 3 3 2 3 0 3 3 3 2 0
 1 2 1 1 1 0 2 3 2 2 0 1 1 3 2 2 0 2 2 3 1 1 0 3 1 3 1 0 2 3 0 2 3 2 3 2 3
 0 3 2 1 0 2 2 2 0 0 1 1 1 3 1 1 2 3 1 2 1 3 2 1 3 2 1 2 3 1 0 2 3 0 1 0 1
 0 2 0 3 0 1 3 2 0 3 2 0 3 0 3 3 0 2 2 2 3 1 2 0 3 2 0 2 3 0 1 0 2 2 2 1 1
 0 3 0 2 0 1 1 0 2 0 2 3 0 0 1 3 1 1 2 1 3 1 1 1 1 1 1 2 3 2 0 1 1 3 0 0 3
 2 0 1 2 0 1 3 1 1 2 0 0 

In [ ]:
cm1=confusion_matrix(y_te,ypred_1)
print(cm1)

[[121   2   0   0]
 [  0 115   3   0]
 [  0   5 127   3]
 [  0   0   1 123]]


In [ ]:
print(classification_report(y_te,ypred_1))

              precision    recall  f1-score   support

           0       1.00      0.98      0.99       123
           1       0.94      0.97      0.96       118
           2       0.97      0.94      0.95       135
           3       0.98      0.99      0.98       124

    accuracy                           0.97       500
   macro avg       0.97      0.97      0.97       500
weighted avg       0.97      0.97      0.97       500



**ACCURACY WITH SVM(LINEAR KERNEL)=0.97**

**PROJECT REPORT**</BR>
Accuracy of Logistic Regression Model=0.63<br>
Accuracy of KNN Classification Model=0.95<br>
Accuracy with SVM(RBF Kernel) Model=0.24<br>
Accuracy with SVM(Linear Kernel) Model=0.97<br>



**THE BEST ACCURACY IS SHOWN BY THE SVM USING LINEAR KERNEL ALGORITHM FOR THE GIVEN PROBLEM STATEMENT**